src/sys/kernel/mod.rs

```diff
@@ -126,6 +129,188 @@ pub fn pid_max() -> ProcResult<i32> {
     read_value("/proc/sys/kernel/pid_max")
 }
 
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+/// Represents the data from `/proc/sys/kernel/sem`
+pub struct SemaphoreLimits {
+    /// The maximum semaphores per semaphore set
+    pub semmsl: u64,
+    /// A system-wide limit on the number of semaphores in all semaphore sets
+    pub semmns: u64,
+    /// The maximum number of operations that may be specified in a semop(2) call
+    pub semopm: u64,
+    /// A system-wide limit on the maximum number of semaphore identifiers
+    pub semmni: u64,
+}
+
+impl SemaphoreLimits {
+    pub fn new() -> ProcResult<Self> {
+        read_value("/proc/sys/kernel/sem")
+    }
+
+    fn from_str(s: &str) -> Result<Self, &'static str> {
+        let mut s = s.split_ascii_whitespace();
+
+        let semmsl = s.next().ok_or("Missing SEMMSL")?;
+        let semmns = s.next().ok_or("Missing SEMMNS")?;
+        let semopm = s.next().ok_or("Missing SEMOPM")?;
+        let semmni = s.next().ok_or("Missing SEMMNI")?;
+
+        let semmsl = semmsl.parse().map_err(|_| "Failed to parse SEMMSL")?;
+        let semmns = semmns.parse().map_err(|_| "Failed to parse SEMMNS")?;
+        let semopm = semopm.parse().map_err(|_| "Failed to parse SEMOPM")?;
+        let semmni = semmni.parse().map_err(|_| "Failed to parse SEMMNI")?;
+
+        Ok(SemaphoreLimits {
+            semmsl,
+            semmns,
+            semopm,
+            semmni,
+        })
+    }
+}
+
+impl FromStr for SemaphoreLimits {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        SemaphoreLimits::from_str(s)
```

Comment:
Instead of forwarding `FromStr::from_str` to `SemaphoreLimits::from_str`, would it be simpler to just have the actual implementation in this method?

Reply:
I think I followed the example of another struct while doing this, though I wouldn't be opposed to just inlining the implementation in the trait impl

Reply:
ok cool.  it's a super minor thing, so let's get this merged and someone can clean this up later 🚀